crates/tests/src/tests/rpc_server/submit_withdrawal_request.rs

```diff
@@ -96,13 +96,15 @@ async fn test_submit_withdrawal_request() {
         .is_request_in_queue(withdrawal_hash)
         .await
         .unwrap();
-    assert!(is_in_queue);
 
-    chain.produce_block(vec![], vec![]).await.unwrap();
+    if !is_in_queue {
+        chain.produce_block(vec![], vec![withdrawal]).await.unwrap();
+    } else {
+        chain.produce_block(vec![], vec![]).await.unwrap();
```

Comment:
What if the withdrawal request remains in the queue for a while? The mem pool state subsequently loaded won't include it, right?

Reply:
Maybe. By default `produce_block` fn will refresh mem pool, it will be included if it's pushed to mem pool.
Add new `produce_block_and_refresh_mem_pool` to allow skip refresh mem pool.

Reply:
My point is that this test is not waiting for the withdrawal to be actually pushed to the mem pool.

I think it should poll until the request is not in queue.

The latest commit still fails if I add a sleep before locking mem pool in line 602 in `registry.rs`.

Reply:
Indeed. Fixed.